evm/test/AccessGrants.js

```diff
@@ -372,4 +372,51 @@ describe("AccessGrants", function () {
       });
     });
   });
+
+  describe("Events", function () {
+    describe("when a grant inserted", function () {
+      it("Creates GrantAdded event", async function () {
+        const { accessGrants, signer1: caller, signer2: owner, signer3: grantee } = await loadFixture(deployAndPopulateContractFixture);
+        const dataId = "849690b7-fee1-46d8-8c91-0268b0cc1850";
+        const lockedUntil = 50n;
+        await accessGrants.connect(owner).insertGrant(grantee, dataId, lockedUntil);
+
+        const eventTopic = ethers.id("GrantAdded(address,address,string,uint256)");
+        const logs = await ethers.provider.getLogs({
+          address: accessGrants.address,
+          topics: [
+            eventTopic,
+            ethers.zeroPadValue(owner.address, 32),
+            ethers.zeroPadValue(grantee.address, 32),
+            ethers.keccak256(ethers.toUtf8Bytes(dataId)),
+          ],
+        });
+        expect(logs.length).to.equal(1);
+        expect(logs[0].data).to.equal(lockedUntil);
```

Comment:
Can we unroll this for clarity/visibility? Something like:
```suggestion
        expect(logs).to.equal([{data: lockedUntil, ...otherStuff}]);
```

If it ends up requiring you to state a bunch of noise, let's drop it.

Reply:
Talked with Alex about this over DMs. Here's how the logs array looks like:
```
[
  Log {
    provider: HardhatEthersProvider {
      _hardhatProvider: [LazyInitializationProviderAdapter],
      _networkName: 'hardhat',
      _blockListeners: [],
      _transactionHashListeners: Map(0) {},
      _eventListeners: []
    },
    transactionHash: '0xc8c14649ea9650528898ce510b09316198792d199e7fcde0b663abf370730ea1',
    blockHash: '0x5ca3322cc9c805927bfa7ce018c11e41f773c62cb9be2ce0c9930f424ee114fa',
    blockNumber: 7,
    removed: false,
    address: '0x5FbDB2315678afecb367f032d93F642f64180aa3',
    data: '0x0000000000000000000000000000000000000000000000000000000000000032',
    topics: [
      '0xaa11b1607ce40799434b358e476f60277b9c8879ca956534f8f185a8b7e227a7',
      '0x0000000000000000000000003c44cdddb6a900fa2b585dd299e03d12fa4293bc',
      '0x00000000000000000000000090f79bf6eb2c4f870365e785982e1f101e93b906',
      '0x2ff2a24fc11ab47b11ff4c5b4f18efa2e0a42debb5cc87a895d20100725157c2'
    ],
    index: 0,
    transactionIndex: 0
  }
]
```

That's way too much noise to have in a test. Let's keep the current code. :+1: